### PR TITLE
fix sync db issues

### PR DESCRIPTION
**Pull request by reybahl**
- fix multiple tags issue in transform function
- fix carriage return issue that prevented database write

### CODE REVIEW

File: config/dev_sync_db.yml [hunk 1]
```diff
@@ -7,4 +7,5 @@ seasons: null
 use_cache: true
 
 sync_db: true
+crawl_evals: false
```
Comment (Josh-Cena) [inline config/dev_sync_db.yml:10]: I think we should remove this

File: .devcontainer/Dockerfile [hunk 2]
```diff
@@ -17,4 +17,4 @@ RUN curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.7/install.sh | b
 
 COPY . .
 
-RUN /root/.cargo/bin/uv pip install --no-cache -e .
+RUN /root/.cargo/bin/uv pip install --no-cache -e . --system
```
Comment (Josh-Cena) [inline .devcontainer/Dockerfile:20]: Any reason why?

File: ferry/database/stage.py [hunk 3]
```diff
@@ -45,6 +45,8 @@ def stage(tables: dict[str, pd.DataFrame], database: Database):
             )
         # create in-memory buffer for DataFrame
         buffer = StringIO()
+        staged_tables[table.name] = staged_tables[table.name].replace(
+            {r'\r': '', r'\n': ' '}, regex=True)
```
Comment (Josh-Cena) [inline ferry/database/stage.py:49]: This looks slightly suspicious to me. Where are the extraneous line breaks coming from and what happens if we remove them?

**Review by Josh-Cena — COMMENTED** [review]
Just a few questions. Maybe @neilsong could also review

File: .devcontainer/Dockerfile (hunk 2 above)
Comment (reybahl) [reply .devcontainer/Dockerfile:20]: this was already merged in my previous PR; not sure why it's still here but Neil made this fix to help set up the Docker environment

File: ferry/database/stage.py (hunk 3 above)
Comment (reybahl) [reply ferry/database/stage.py:49]: This code actually also works without removing the occurrences of `\n`. Would it be better if I made that change?

This is the error I was getting earlier:

`psycopg2.errors.BadCopyFileFormat: literal carriage return found in data`

I can try to look more into the cause of the occurrences of `\r`, but as of right now, I'm not sure where they are coming from and if we are losing any information by removing them.

**Comment by reybahl** [comment]
This is a snippet of one of the string buffers for writing to database: 

```
Please rate the following aspects of your experience in this course or workshop regarding: the Instructor.\r\\\n\r\\\n Preparation Ratings
```

Not sure what the purpose of \r in this case is; is it important to keep them here?

**Comment by Josh-Cena** [comment]
I'm very surprised it exists here and no I don't think it should exist in our dumped tables.

File: .devcontainer/Dockerfile (hunk 2 above)
Comment (neilsong) [reply .devcontainer/Dockerfile:20]: yea that shouldn't show up as a diff? odd

File: ferry/database/stage.py [hunk 4]
```diff
@@ -45,6 +45,8 @@ def stage(tables: dict[str, pd.DataFrame], database: Database):
             )
         # create in-memory buffer for DataFrame
         buffer = StringIO()
+        staged_tables[table.name] = staged_tables[table.name].replace(
```
Comment (Josh-Cena) [inline ferry/database/stage.py:48]:
```suggestion
        # TODO is this really needed?
        staged_tables[table.name] = staged_tables[table.name].replace(
```

**Review by Josh-Cena — APPROVED** [review]
I think this is good to merge. Thanks!

**Pull request merged**